File: src/main/python/net/coderextreme/json/TextExamples.py
from jnius import autoclass
from X3Dautoclass import *
X3D0 =  X3DObject().setProfile("Immersive").setVersion("3.0")
head1 = headObject()
meta2 = metaObject().setName("title").setContent("TextExamples.x3d")
head1.addMeta(meta2)
meta3 = metaObject().setName("description").setContent("Show different escape-character text examples for embedded quotation marks.")
head1.addMeta(meta3)
meta4 = metaObject().setName("creator").setContent("Don Brutzman")
head1.addMeta(meta4)
meta5 = metaObject().setName("created").setContent("7 April 2001")
head1.addMeta(meta5)
meta6 = metaObject().setName("modified").setContent("26 April 2016")
head1.addMeta(meta6)
meta7 = metaObject().setName("warning").setContent("Note that X3D Canonicalization (C14N) will scrub alternate XML character representations, be careful to check original encoding into version control.")
head1.addMeta(meta7)
meta8 = metaObject().setName("warning").setContent("Usually this source document needs to be inspected and edited using a plain-text editor in order to see the differences in these XML-equivalent text representations.")
head1.addMeta(meta8)
meta9 = metaObject().setName("identifier").setContent("http://www.web3d.org/x3d/content/examples/Basic/development/TextExamples.x3d")
head1.addMeta(meta9)
meta10 = metaObject().setName("generator").setContent("X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit")
head1.addMeta(meta10)
meta11 = metaObject().setName("license").setContent("../license.html")
head1.addMeta(meta11)
X3D0.setHead(head1)
Scene12 = SceneObject()
Transform13 = TransformObject().setTranslation([0,2,0])
Shape14 = ShapeObject()
Text15 = TextObject().setString(["Compare special character escaping"])
FontStyle16 = FontStyleObject().setDEF("testFontStyle").setJustify(["MIDDLE","MIDDLE"]).setSize(0.8)
Text15.setFontStyle(FontStyle16)
Shape14.setGeometry(Text15)
Appearance17 = AppearanceObject().setDEF("LightBlueAppearance")
Material18 = MaterialObject().setDiffuseColor([0.1,0.7,0.7])
Appearance17.setMaterial(Material18)
Shape14.setAppearance(Appearance17)
Transform13.addChild(Shape14)
Scene12.addChild(Transform13)
Transform19 = TransformObject().setTranslation([-3,0,0])
Shape20 = ShapeObject()
Text21 = TextObject().setString(["I don't think so","","he said \"Hi\""])
FontStyle22 = FontStyleObject().setUSE("testFontStyle")
Text21.setFontStyle(FontStyle22)
Shape20.setGeometry(Text21)
Appearance23 = AppearanceObject().setUSE("LightBlueAppearance")
Shape20.setAppearance(Appearance23)
Transform19.addChild(Shape20)
Scene12.addChild(Transform19)
Transform24 = TransformObject().setTranslation([3,0,0])
Shape25 = ShapeObject()
Text26 = TextObject().setString(["I don't think so","","he said \"Hi\""])
FontStyle27 = FontStyleObject().setUSE("testFontStyle")
Text26.setFontStyle(FontStyle27)
Shape25.setGeometry(Text26)
Appearance28 = AppearanceObject().setUSE("LightBlueAppearance")
Shape25.setAppearance(Appearance28)
Transform24.addChild(Shape25)
Scene12.addChild(Transform24)
X3D0.setScene(Scene12)

X3D0.toFileX3D("../new/json/TextExamples.new.x3d")

File: src/main/java/net/x3djsonld/data/TextExamples.java
package net.x3djsonld.data;

import java.util.*;
import org.web3d.x3d.jsail.Core.*;
import org.web3d.x3d.jsail.fields.*;
import org.web3d.x3d.jsail.Grouping.*;
import org.web3d.x3d.jsail.Shape.*;
import org.web3d.x3d.jsail.Text.*;

// Javadoc annotations follow, see below for source.
/**
 * <p> Show different escape-character text examples for embedded quotation marks. </p>
 <p> Related links: <a href="../../../development/TextExamples.java">TextExamples.java</a> source, <a href="../../../development/TextExamplesIndex.html" target="_top">TextExamples catalog page</a>, <a href="http://www.web3d.org/x3d/content/examples/X3dResources.html" target="_blank">X3D Resources</a>, <a href="http://www.web3d.org/x3d/content/examples/X3dSceneAuthoringHints.html" target="_blank">X3D Scene Authoring Hints</a> and <a href="http://www.web3d.org/x3d/content/X3dTooltips.html" target="_blank">X3D Tooltips</a>. </p>
	<table style="color:black; border:0px solid; border-spacing:10px 0px;" summary="Scene Metadata">
		<tr style="background-color:silver; border-color:silver;">
			<td style="text-align:center; padding:10px 0px;"><i>meta tags</i></td>
			<td style="text-align:left;   padding:10px 0px;">&nbsp; Document Metadata </td>
		</tr>

		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> title </i> </td>
			<td> <a href="../../../development/TextExamples.x3d">TextExamples.x3d</a> </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> description </i> </td>
			<td> Show different escape-character text examples for embedded quotation marks. </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> creator </i> </td>
			<td> Don Brutzman </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> created </i> </td>
			<td> 7 April 2001 </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> modified </i> </td>
			<td> 26 April 2016 </td>
		</tr>
		<tr style="color:burntorange">
			<td style="text-align:right; vertical-align: text-top;"> <i> warning </i> </td>
			<td> Note that X3D Canonicalization (C14N) will scrub alternate XML character representations, be careful to check original encoding into version control. </td>
		</tr>
		<tr style="color:burntorange">
			<td style="text-align:right; vertical-align: text-top;"> <i> warning </i> </td>
			<td> Usually this source document needs to be inspected and edited using a plain-text editor in order to see the differences in these XML-equivalent text representations. </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> identifier </i> </td>
			<td> <a href="http://www.web3d.org/x3d/content/examples/Basic/development/TextExamples.x3d" target="_blank">http://www.web3d.org/x3d/content/examples/Basic/development/TextExamples.x3d</a> </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> generator </i> </td>
			<td> X3D-Edit 3.3, <a href="https://savage.nps.edu/X3D-Edit" target="_blank">https://savage.nps.edu/X3D-Edit</a> </td>
		</tr>
		<tr>
			<td style="text-align:right; vertical-align: text-top;"> <i> license </i> </td>
			<td> <a href="../../../development/../license.html">../license.html</a> </td>
		</tr>
		<tr style="background-color:silver; border-color:silver;">
			<td style="text-align:center;" colspan="2">  &nbsp; </td>
		</tr>
	</table>

	<p>
		This program uses the
		<a href="http://www.web3d.org/specifications/java/X3DJSAIL.html" target="_blank">X3D Java Scene Access Interface Library (X3DJSAIL)</a>.
		It has been produced using the 
		<a href="http://www.web3d.org/x3d/stylesheets/X3dToJava.xslt" target="_blank">X3dToJava.xslt</a>
		stylesheet to create Java source code from an <code>.x3d</code> scene.
	</p>

	* @author Don Brutzman
 */

public class TextExamples
{
  /** Default constructor to create this object. */
  public TextExamples ()
  {
    initialize();
  }
	
  /** Create and initialize the X3D model. */
  public final void initialize()
  {
  x3dModel = new X3DObject().setProfile("Immersive").setVersion("3.0")
  .setHead(new headObject()
    .addMeta(new metaObject().setName("title").setContent("TextExamples.x3d"))
    .addMeta(new metaObject().setName("description").setContent("Show different escape-character text examples for embedded quotation marks."))
    .addMeta(new metaObject().setName("creator").setContent("Don Brutzman"))
    .addMeta(new metaObject().setName("created").setContent("7 April 2001"))
    .addMeta(new metaObject().setName("modified").setContent("26 April 2016"))
    .addMeta(new metaObject().setName("warning").setContent("Note that X3D Canonicalization (C14N) will scrub alternate XML character representations, be careful to check original encoding into version control."))
    .addMeta(new metaObject().setName("warning").setContent("Usually this source document needs to be inspected and edited using a plain-text editor in order to see the differences in these XML-equivalent text representations."))
    .addMeta(new metaObject().setName("identifier").setContent("http://www.web3d.org/x3d/content/examples/Basic/development/TextExamples.x3d"))
    .addMeta(new metaObject().setName("generator").setContent("X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit"))
    .addMeta(new metaObject().setName("license").setContent("../license.html")))
  .setScene(new SceneObject()
    .addChild(new TransformObject().setTranslation(0.0f,2.0f,0.0f)
      .addChild(new ShapeObject()
        .setGeometry(new TextObject().setString(new MFStringObject("\"Compare special character escaping\""))
          .setFontStyle(new FontStyleObject("testFontStyle").setJustify(new MFStringObject("\"MIDDLE\" \"MIDDLE\"")).setSize(0.8f)))
        .setAppearance(new AppearanceObject("LightBlueAppearance")
          .setMaterial(new MaterialObject().setDiffuseColor(0.1f,0.7f,0.7f)))))
    .addChild(new TransformObject().setTranslation(-3.0f,0.0f,0.0f)
      .addChild(new ShapeObject()
        .setGeometry(new TextObject().setString(new MFStringObject("\"I don't think so\" \"\" \"he said \\\"Hi\\\"\""))
          .setFontStyle(new FontStyleObject().setUSE("testFontStyle")))
        .setAppearance(new AppearanceObject().setUSE("LightBlueAppearance"))))
    .addChild(new TransformObject().setTranslation(3.0f,0.0f,0.0f)
      .addChild(new ShapeObject()
        .setGeometry(new TextObject().setString(new MFStringObject("\"I don't think so\" \"\" \"he said \\\"Hi\\\"\""))
          .setFontStyle(new FontStyleObject().setUSE("testFontStyle")))
        .setAppearance(new AppearanceObject().setUSE("LightBlueAppearance")))));
  }
  // end of initialize() method

  /** The initialized model object, created within initialize() method. */
  private X3DObject x3dModel;
  
  /** Provide a shallow copy of the X3D model.
   * @return TextExamples model
   */
  public X3DObject getX3dModel()
  {	  
	  return x3dModel;
  }
  
  /** Indicate X3DJSAIL validation results for this X3D model.
   * @return validation results plus exception information, if any
   */
  public String validateSelf()
  {
	String       metaResult = new String();
	String validationResult = new String();
	String  exceptionResult = new String();
	try
	{
		initialize();
		
		if ((getX3dModel() == null) || (getX3dModel().getHead() == null))
		{
			validationResult = "empty scene, nothing to validate. " + x3dModel.validate();
			return validationResult;
		}
		// first list informational meta elements of interest
		for (metaObject meta : getX3dModel().getHead().getMetaList())
		{
			if (meta.getName().equals(metaObject.NAME_ERROR) ||
				meta.getName().equals(metaObject.NAME_WARNING) ||
				meta.getName().equals(metaObject.NAME_HINT) ||
				meta.getName().equals(metaObject.NAME_INFO) ||
				meta.getName().equals(metaObject.NAME_TODO))
			{
				metaResult += meta.toStringX3D();
			}
		}
		validationResult += x3dModel.validate(); // walk entire tree to validate correctness
	}
	catch (Exception e)
	{
		exceptionResult = e.getMessage(); // report exception failures, if any
	}
	if  (metaResult.isEmpty() && exceptionResult.isEmpty() && validationResult.isEmpty())
	     return "success";
	else
	{
		String returnMessage = metaResult;
		if  (!exceptionResult.isEmpty() && !validationResult.isEmpty())
			returnMessage += "\n*** ";
		returnMessage += exceptionResult;
		if  (exceptionResult.isEmpty() && !validationResult.isEmpty())
			returnMessage = "\n" + returnMessage; // skip line before meta tags, etc.
		returnMessage += validationResult;
		return returnMessage;
	}
  }
    /** Default main() method provided for test purposes.
     * @param argv input parameters
     */
    public static void main(String argv[])
    {
		TextExamples testObject = new TextExamples();
		System.out.println ("TextExamples execution self-validation test results: " + testObject.validateSelf());
	}
}